### PR TITLE
Push send page

### PR DESCRIPTION
**Pull request by drew-gross**
Push Console! Most of this code is horrible, due to us building, destroying, and rebuilding the Parse.Query used for installation targeting so many times and in so many ways. Eventually we should just have a Parse.Query everywhere.

### CODE REVIEW

**Comment by flovilmart** [comment]
HOLY MOLLY!

**Comment by peterdotjs** [comment]
awesome! push lives!

File: src/components/PushAudienceDialog/PushAudienceDialog.react.js [hunk 1]
```diff
@@ -27,6 +27,9 @@ import TextInput             from 'components/TextInput/TextInput.react';
 import Toggle                from 'components/Toggle/Toggle.react';
 import { List, Map }         from 'immutable';
 
+const PARSE_SERVER_SUPPORTS_SAVED_AUDIENCES = false;
```
Comment (peterdotjs) [inline src/components/PushAudienceDialog/PushAudienceDialog.react.js]: will this eventually be read from serverInfo?

File: src/dashboard/Push/PushNew.react.js [hunk 2]
```diff
@@ -182,13 +197,26 @@ export default class PushNew extends DashboardView {
 
   handlePushSubmit(changes) {
     let promise = new Promise();
-    this.context.currentApp.createPushNotification(changes).then(({ error }) => {
+    let payload = {};
+    payload.alert = changes.data_type === 'json' ? JSON.parse(changes.data) : changes.data;
+    payload.badge = !!changes.increment_badge;
+    Parse.Push.send({
+      where: changes.target ? changes.target : new Parse.Query(Parse.Installation),
```
Comment (peterdotjs) [inline src/dashboard/Push/PushNew.react.js]: `changes.target || new Parse.Query(Parse.Installation)`

File: src/components/PushAudienceDialog/PushAudienceDialog.react.js (hunk 1 above)
Comment (drew-gross) [inline src/components/PushAudienceDialog/PushAudienceDialog.react.js]: Yep thats that plan

File: src/dashboard/Push/PushNew.react.js (hunk 2 above)
Comment (drew-gross) [inline src/dashboard/Push/PushNew.react.js]: Will do

**Comment by chrkaatz** [comment]
Nice!

File: src/dashboard/Dashboard.js [hunk 3]
```diff
@@ -226,7 +226,7 @@ class Dashboard extends React.Component {
           <Route path='config' component={Config} />
           <Route path='api_console' component={ApiConsole} />
           <Route path='migration' component={Migration} />
-          <Redirect from='push' to='/apps/:appId/push/activity/all' />
+          <Redirect from='push' to='/apps/:appId/push/new' />
```
Comment (peterdotjs) [inline src/dashboard/Dashboard.js]: are we not planning to have the index page? Why not have empty index page for now and start creation flow with create new?

Comment (drew-gross) [inline src/dashboard/Dashboard.js]: Push status aren't saved on the server yet so there is no data to load. Once we have some data to load, we'll add the index page back. I figured this was a find temporary solution.

Comment (peterdotjs) [inline src/dashboard/Dashboard.js]: fair enough

**Comment by Webde** [comment]
Your version #192 no run in my project. it show in terminal `Uncaught SyntaxError: Unexpected token <     dashboard.bundle.js:1`

**Comment by Webde** [comment]
In this moment is not posible see in my Dashboard the push UI, Help me please

**Pull request merged**